src/pykeen/nn/pyg.py

```diff
@@ -196,34 +202,47 @@ def __init__(
         # buffer edge index for message passing
         self.register_buffer(name="edge_index", tensor=triples_factory.mapped_triples[:, [0, 2]].t())
 
+        self.restrict_k_hop = restrict_k_hop
+
         # TODO: inductiveness; we need to
         #   * replace edge_index
         #   * replace base representations
         #   * keep layers & activations
 
     # docstr-coverage: inherited
     def _plain_forward(self, indices: Optional[torch.LongTensor] = None) -> torch.FloatTensor:  # noqa: D102
-        # TODO: we could reduce the memory footprint and maybe also computation time
-        #       by considering only the k-hop neighborhood of the requested indices
-        #       computing this may be computationally expensive, too
-        # get *all* base representations
-        x = self.base(indices=None)
-        # perform message passing on *all* base representations & edges
-        x = self.pass_messages(x=x)
+        if indices is not None and self.restrict_k_hop:
+            assert k_hop_subgraph is not None
+            # (1) the nodes involved in the subgraph
+            # (2) the filtered edge_index connectivity
+            # (3) the mapping from node indices in node_idx to their new location, and
+            # (4) the edge mask indicating which edges were preserved
+            neighbor_indices, _, indices, edge_mask = k_hop_subgraph(
+                node_idx=indices, num_hops=len(self.layers), edge_index=self.edge_index, relabel_nodes=True
+            )
+            x = self.base(indices=neighbor_indices)
```

Comment:
I'd be very careful with `relabel_nodes`  - it creates a new edge index with new node IDs, and the features in `x` must adhere to the new mapping (re-labeling)

Reply:
That is what we want to have, right? Because we do not want to calculate *all* base representations but only those corresponding to the neighbors' indices (e.g., because we use expensive base representations such as a Transformer) -> we need to adapt the `edge_index`

Reply:
I added a test to check for consistent results with and without `k_hop_subgraph`: e8591ce